src/attr/_make.py

```diff
@@ -529,6 +529,24 @@ def _patch_original_class(self):
         for name, value in self._cls_dict.items():
             setattr(cls, name, value)
 
+        # attach __setstate__. This is necessary to clear the hash code
+        # cache on deserialization. See issue
+        # https://github.com/python-attrs/attrs/issues/482 .
+        # Note that this code only handles setstate for non-slots classes.
+        # For slots classes, see similar code in _create_slots_class .
+        if self._cache_hash:
+            if hasattr(cls, "__setstate__"):
+                raise NotImplementedError(
```

Comment:
I guess this should be marked as a breaking change? 🤔 It should be mentioned that this isn't possible anymore in the newsfragment.